### PR TITLE
Fix notify(Parent/Child)ItemChanged methods

### LINKED ISSUE #94: notifyChildItemChanged
notifyChildItemChanged causes notifyItemChanged. I would like to have a notifyDataSetChanged method which cause a onItemsChanged for the child.  snx)

### PR DESCRIPTION
**Pull request by paul-turner**
fixes #94 
We were not fetching the updated children objects which meant that new references were not being caught.

### CODE REVIEW

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java [hunk 1]
```diff
@@ -742,8 +743,11 @@ public void notifyParentItemChanged(int parentPosition) {
         parentWrapper.setParentListItem(parentListItem);
         int sizeChanged = 1;
         if (parentWrapper.isExpanded()) {
-            int childListSize = parentWrapper.getChildItemList().size();
+            List<?> childItems = parentWrapper.getChildItemList();
+            int childListSize = childItems.size();
             for (int i = 0; i < childListSize; i++) {
+                Object child = childItems.get(i);
```
Comment (dgreenhalgh) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: Allocation in a loop

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java [hunk 2]
```diff
@@ -821,7 +825,10 @@ public void notifyChildItemChanged(int parentPosition, int childPosition) {
         ParentWrapper parentWrapper = (ParentWrapper) mItemList.get(parentWrapperIndex);
         parentWrapper.setParentListItem(parentListItem);
         if (parentWrapper.isExpanded()) {
-            notifyItemChanged(parentWrapperIndex + childPosition + 1);
+            int listChildPosition = parentWrapperIndex + childPosition + 1;
+            Object child = parentWrapper.getChildItemList().get(childPosition);
```
Comment (dgreenhalgh) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: Allocation in a loop

**Comment by paul-turner** [comment]
Fixed

**Comment by dgreenhalgh** [comment]
Needs re-up

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java (hunk 2 above)
Comment (paul-turner) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: This one was not in a loop so did not change.

**Comment by paul-turner** [comment]
should be good to go

**Pull request merged**